lib/x86simdsort.cpp

```diff
@@ -10,7 +10,7 @@ static int check_cpu_feature_support(std::string_view cpufeature)
     const char *disable_avx512 = std::getenv("XSS_DISABLE_AVX512");
 
     if ((cpufeature == "avx512_spr") && (!disable_avx512))
-#if defined(__FLT16_MAX__) && !defined(__INTEL_LLVM_COMPILER)
+#if defined(__FLT16_MAX__) && !defined(__INTEL_LLVM_COMPILER) && !(defined(__OpenBSD__) &&  __clang_major__ < 18)
```

Comment:
This particular build fail is not related to OpenBSD, it is related to just clang version. 
```suggestion
#if defined(__FLT16_MAX__) && !defined(__INTEL_LLVM_COMPILER) &&  __clang_major__ >= 18
```